src/main/webapp/resources/js/utilities/antdesign-table-utilities.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * This file is for any dynamic settings required for an ant design table.
+ */
+
+/**
+ * Default page size for ant design tables
+ */
+export const defaultPageSize = 10;
+
+export const defaultShowPageSizeChanger = true;
+
+/**
+ * Set the common pagination options for the tables
+ */
+export function getPaginationOptions(totalEntries) {
+  if (totalEntries <= 10) {
+    return {
+      pageSizeOptions: ["10"],
+      hideOnSinglePage: true,
+      showSizeChanger: false,
+      pageSize: defaultPageSize,
+    };
+  } else if (totalEntries <= 20) {
+    return {
+      pageSizeOptions: ["10", "20"],
+      hideOnSinglePage: totalEntries <= defaultPageSize,
+      showSizeChanger: defaultShowPageSizeChanger,
+      pageSize: defaultPageSize,
+    };
+  } else if (totalEntries <= 50) {
+    return {
+      pageSizeOptions: ["10", "20", "50"],
+      hideOnSinglePage: totalEntries <= defaultPageSize,
+      showSizeChanger: defaultShowPageSizeChanger,
+      pageSize: defaultPageSize,
+    };
+  } else {
+    return {
+      pageSizeOptions: ["10", "20", "50", "100"],
+      hideOnSinglePage: totalEntries <= defaultPageSize,
+      showSizeChanger: defaultShowPageSizeChanger,
+      pageSize: defaultPageSize,
+    };
+  }
```

Comment:
I think you should simplify this to:
```javascript
  const config = {
    hideOnSinglePage: totalEntries <= defaultPageSize,
    showSizeChanger: defaultShowPageSizeChanger,
    pageSize: defaultPageSize,
  };

  if (totalEntries <= 10) {
    return { ...config, pageSizeOptions: [10], showSizeChanger: false };
  } else if (totalEntries <= 20) {
    return { ...config, pageSizeOptions: [10, 20] };
  } else if (totalEntries <= 50) {
    return {
      ...config,
      pageSizeOptions: [10, 20, 50],
    };
  } else {
    return {
      ...config,
      pageSizeOptions: [10, 20, 50, 100],
    };
  }
  ```

Reply:
Updated in c96bd77d27718c9064e6535c7aac61dd11b3f80b